src/deployment/registration.py

```diff
@@ -235,7 +235,10 @@ def add_application_password(app_object_id: UUID) -> Tuple[str, str]:
             return add_application_password_impl(app_object_id)
         except GraphQueryError as err:
             error = err
-            logging.warning("unable to create app password: %s", err.message)
+            if "Request_ResourceNotFound" in repr(err):
```

Comment:
Can we cite something for the origin of this search substring? Or at least comment on where it is empirically derived?

Same applies to `PrincipalNotFound`, above.

Reply:
I added some refs to how this is similar to az-cli